packages/bsky/src/services/indexing/index.ts

```diff
@@ -93,7 +97,20 @@ export class IndexingService {
       const indexer = indexingTx.findIndexerForCollection(uri.collection)
       if (!indexer) return
       if (action === WriteOpAction.Create) {
-        await indexer.insertRecord(uri, cid, obj, timestamp)
+        const insertedRecords = await indexer.insertRecord(
+          uri,
+          cid,
+          obj,
+          timestamp,
+        )
+        // THIS WILL SEND THE NOTIFICATIONS
+        if (insertedRecords) {
+          /* don't await */ this.notifServer
```

Comment:
for things we don't want to do synchronously, we normally throw them on the `backgroundQueue` (which is available in this service)

that let's us wait for backgrounded tasks to finish before tearing down an instance, puts a cap on concurrency of backgrounded tasks & it handles error handling + logging

Reply:
you can see an example of this in, for instance, the `Labeler` class

it generally looks something like this:
```ts
    this.backgroundQueue.add(() =>
      this.createAndStoreLabels(uri, obj).catch((err) => {
        log.error(
          { err, uri: uri.toString(), record: obj },
          'failed to label record',
        )
      }),
    )
```

Reply:
Done